app/com/arpnetworking/commons/akka/JacksonSerializer.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.commons.akka;
+
+import akka.actor.ExtendedActorSystem;
+import akka.serialization.JSerializer;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import java.io.IOException;
+
+/**
+ * Serializer for Akka using Jackson.
+ * <br>
+ * This should be replaced with akka-jackson-serialization once this project is
+ * is on Scala 2.12+ and Akka 2.6+.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class JacksonSerializer extends JSerializer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JacksonSerializer.class);
+    private static @Nullable ObjectMapper gObjectMapper;
+
+    /**
+     * Constructor used by Akka upon system initialization.
+     *
+     * @param ignored the actor system
+     */
+    public JacksonSerializer(final ExtendedActorSystem ignored) {}
```

Comment:
Is there no way to inject the `ObjectMapper` instance into the serializer? The current mutable global static seems problematic.

Reply:
While I agree and don't like it either, I've been unable to find any other method of injection. The Akka docs suggest that the only way is to add a custom section into the `.conf` file and load it from that, but I don't want to duplicate the ObjectMapper setup.

[This stack overflow user](https://stackoverflow.com/questions/57050458/akka-add-custom-serializer-programmatically) also tried and failed to find a way to do serializer registration at runtime.